util/doc.config

```diff
@@ -63,6 +63,7 @@
   },{
     "Name": "SSL implementation",
     "Headers": [
+      "include/openssl/opensslconf.h",
```

Comment:
Does this belong under "SSL implementation"? I'm not sure which of the other categories would make more sense though...

Reply:
that's a good question, `opensslconf.h`seems to span both `libcrypto` and `libssl`. what do you think @andrewhop?

Reply:
hm, so it looks like available sections are

- Low-level infrastructure
- Low-level crypto primitives
- Crypto interfaces
- Legacy ASN.1 and X.509 implementation
- SSL implementation

`opensslconf.h` contains configuration macros pertaining to SSL/TLS, low-level primitives, obscure protocols (e.g. trusted timestamping), etc. it's a bit of a grab bag, but I think out of the available sections "SSL implementation" is most appropriate.

I suppose we could also break it out into a new "configuration" section.

Reply:
This seems fine for now, thanks for getting this in.